src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.cs

```diff
@@ -206,6 +199,9 @@ private async Task ProcessQueueAsync()
                 {
                     var work = await _queue.DequeueAsync().ConfigureAwait(false);
 
+                    // Record when the work item was been de-queued and the request context preparation started.
+                    work.Metrics.RecordExecutionStart();
```

Comment:
There is a _teensy_ risk here that if the work item never starts, this will get lost, but personally I think thats fine. The other logging should cover that scenario, and I'd worry that recording the exception below would make it look like this method was the culprit, which would be a red herring.

Just thought I'd mention it though, in case someone disagrees. Moving this call to the callback in the QueueItem would probably make this look a bit nicer, because there wouldn't be a gap, but we'd lose timing on how long it takes to construct the LSP solution, so ¯\\\_(ツ)_/¯

Reply:
The view I had was that if we throw an exception during the non-request handling bits of the queue processing (basically just creating the request context) then the queue will be shutting down anyway and we'll get a fatal error.  So it's kind of like the request was never handled at all so no point in recording.